src/cargo/core/compiler/future_incompat.rs

```diff
@@ -236,7 +236,7 @@ fn render_report(per_package_reports: &[FutureIncompatReportPackage]) -> BTreeMa
     let mut report: BTreeMap<String, String> = BTreeMap::new();
     for per_package in per_package_reports {
         let package_spec = format!(
-            "{}:{}",
+            "{}@{}",
             per_package.package_id.name(),
             per_package.package_id.version()
         );
```

Comment:
And with the following line:
```rust
let rendered = report.entry(package_spec).or_default();
```
This changes how the reports are created on disk.  The `report future-incompat --package` flag doesn't take a package-spec but whatever was put into the report here.

- We could just keep this as `:` but then `report future-incompat` will only accept `:`
- We could have `report future-incompat` try to generate both formats

Any thoughts?  @joshtriplett since I think you had worked on this

Reply:
@epage We discussed this in the Cargo meeting; summary: go ahead and change it, don't worry about `:` at all. There's a format version number that you could bump so that old Cargo doesn't think it can read this.

Reply:
I've not dug further but from what I've seen of the code and failure modes, I don't think we even need to bump the format version number.  It looks like `report future-incompat` is completely neutral to the pkgid spec format.  It accepts a parameter and uses it as a key.  You should be able to have old cargo's read the file.